### ISSUE TITLE
(Menu) History List Size also limits Favorites playlist

### ISSUE THREAD

**Issue by vgambit** [libretro/RetroArch#9098]
## Description

The History List Size setting in Settings > Playlists limits the **Favorites** playlist _in addition to_ the **History** playlist.

### Expected behavior

I spend a couple of hours scrolling through a long list of games and adding the ones I'm interested in to the Favorites playlist. I end up adding around 150 or so, from A to Z. I go to the Favorites list, and it has all the games I added, sorted alphabetically. Essentially, I wouldn't expect the Favorites list to have a maximum value. If anything, I would have that maximum be its own value, separate from the History maximum, and set to 999 by default. I would also expect it to notify me if I attempt to add a game to that playlist while the list is already full.

### Actual behavior

After checking the playlist, I notice that the list starts at around N instead of A. I scroll through, counting the games, and it's 110 (which is more than the History List Size maxium). I add more games and count again, and the count is 99 (which is the default maximum).

### Steps to reproduce the bug

1. Set the History List Size value to a low value
2. Add a number of games to your Favorites playlist greater than the History List Size
3. Observe that your Favorites list does not contain all the games you added to it

### Bisect Results

Basically, at some point after I hit the History List Size default (99), entries started getting removed from my Favorites list without notification (beyond "Added to Favorites"), on a rolling basis like you'd expect from the History playlist.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.7.7/3853d89

### Environment information

- OS: Switch/NX

**Comment by twinaphex** [comment]
Hi there @jdgleaver,

could you look into this?

**Comment by jdgleaver** [comment]
@twinaphex Sure, no problem.

I'm a little tied up at the moment (whole week's been chaos, actually...), but sometime on Monday I can certainly add a new 'favourites size' option, and have a notification displayed if the user tries to exceed it. Shouldn't be any trouble!

**Comment by jdgleaver** [comment]
Okay, PR #9229 should fix this  :)
